splink/profile_data.py

```diff
@@ -159,21 +165,52 @@ def _col_or_expr_frequencies_raw_data_sql(cols_or_exprs, table_name):
                 end
             """
 
-        sql = f"""
-        select * from
-        (select
-            count(*) as value_count,
-            '{gn}' as group_name,
-            cast({col_or_expr} as varchar) as value,
-            (select count({col_or_expr}) from {table_name}) as total_non_null_rows,
-            (select count(*) from {table_name}) as total_rows_inc_nulls,
-            (select count(distinct {col_or_expr}) from {table_name})
-                as distinct_value_count
-        from {table_name}
-        where {col_or_expr} is not null
-        group by {col_or_expr}
-        order by count(*) desc) column_stats
-        """
+        if not cast_arrays_as_str and raw_expr in array_cols:
+
+            sql = f"""
+                select * from
+                (select value,
+                count (*) as value_count,
+                '{gn}' as group_name,
+
+                (select count(value) from
+                (select unnest ({col_or_expr}
+                ) AS value from {table_name})) as total_non_null_rows,
+
+                (select count(*) from
+                (select unnest ({col_or_expr}
+                ) AS value from {table_name})) as total_rows_inc_nulls,
+
+                (select count(distinct value) from
+                (select unnest ({col_or_expr}
+                ) AS value from {table_name})) as distinct_value_count
+
+                from
+                (select cast(unnest({col_or_expr}) as varchar) as value,
+
+                from {table_name})
+                group by value
+                order by count(*) desc)
```

Comment:
would it be possible to name this subquery (e.g. -> `order by count(*) desc) column_stats`) as in the current SQL (sorry, I know this change has appeared more recently than when you started this branch!) - this is needed for the query to run in Postgres